Source/Core/Core/CoreParameter.h

```diff
@@ -201,6 +204,15 @@ struct SCoreStartupParameter
 	int iHotkey[NUM_HOTKEYS];
 	int iHotkeyModifier[NUM_HOTKEYS];
 
+	//Remove Layer
+	std::vector<SkipEntry> object_removal_codes;
+	u32 skip_objects_end = 0;
+	u32 skip_objects_start = 0;
+	size_t num_object_removal_codes = 0;
+	std::vector<int> num_object_removal_data_bytes;
+	volatile bool update = true;
+	volatile bool done = true;
```

Comment:
I think a mutex is maybe a bit slower than using a volatile, but the volatile isn't safe either. Depending on the comiler, they reorder memory access over the volatile. So for a real synchronnization, you'd have to use atomics which are often even slower than mutexes.
Also if we start thinking about non-coherent cpus (no x86 or arm), the volatile won't provide visibility, so the locking will just be broken.

But the main argument here: It doesn't matter at all. Both volatile, mutex and atomics are fast enough to not notice any difference in this use case. Our fifo + opcode decoder + vertex loader are just mutch slower...